### ISSUE TITLE
Implement "Close Tabs Below" in the script editor

### ISSUE THREAD

**Issue by Jamsers** [godotengine/godot-proposals#5976]
### Describe the project you are working on

Any project using GDScript.

### Describe the problem or limitation you are having in your project

I open a lot of documentation tabs in the script editor (which are mixed in with the script tabs) and would like a convenient way to close all documentation tabs while leaving the script tabs open. I would also like a way to keep a whole bunch of tabs and close all other tabs - right now, "Close Other Tabs" means you can only really keep one tab open, unless you close each tab you want closed individually.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The script editor already seems to automatically put all documentation tabs below the script tabs, and you can already freely change the tab order with drag and drop. With the addition of "Close Tabs Below" functionality, it should make keeping a specific group of tabs while closing all others easy.

This functionality is similar to "Close tabs to the right" in web browsers.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

It'll be a button added to the menu opened when you right click on a tab, labelled "Close Tabs Below", and it'll close all tabs below the tab you right clicked on, when selected.

![Screenshot (12)](https://user-images.githubusercontent.com/39361911/209087005-a7a33f59-3c45-4fd2-8e8c-10c7978400fe.png)

(Apologies for the typo 😅)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I'm not sure, I haven't tried extending the editor with scripting yet.

### Is there a reason why this should be core and not an add-on in the asset library?

It seems like a relatively simple addition to the script editor, so it's probably overkill to implement it in a separate asset/plugin.

**Comment by TheDuriel** [comment]
In the Script Editor. Click "File", then read the options presented to you.

You can even bind a hotkey to the "Close Docs" option if you wanted.

**Comment by Jamsers** [comment]
>In the Script Editor. Click "File", then read the options presented to you.
>You can even bind a hotkey to the "Close Docs" option if you wanted.

"Close Docs" solves the closing all documentation tabs use case, but doesn't solve the keeping a specific group of tabs open while closing all others use case.

**Comment by TheDuriel** [comment]
That's not going to be solved by a "close all below" button either though. You'd need more sophisticated means of organizing the script list, with tab groups.

Try using the FileDock to browse scripts and quickly jump between topics? You can type .gd in to the search to filter.

**Comment by Jamsers** [comment]
>That's not going to be solved by a "close all below" button either though.

It would, you just drag and drop all the tabs you want to keep open, to the top, then "Close All Below" on the last tab.
Basically this, but imagine it's vertical:

https://user-images.githubusercontent.com/39361911/209150235-be095ceb-a96a-4d7f-9781-08eb2bb940ac.mp4

>You'd need more sophisticated means of organizing the script list, with tab groups.

I wouldn't be opposed to this as well, this would be the ultimate solution.

**Comment by TheDuriel** [comment]
If you are already dragging and dropping tabs. You can also ctrl+w the tab while you're doing it.

While I don't see any point in it, because the File Dock serves that purpose. Seeing what better script tab organization could look like in a proposal could be nice.

Godot has a problem with context menus being bloated though.

**Comment by DissonantVoid** [comment]
maybe instead we can allow the user to shift click to select a bunch of scripts, then close them all at the same time? it doesn't seem like we can select multiple scripts at once so maybe that would come in handy

**Comment by KoBeWi** [comment]
This proposal is basically equivalent of "Close Tabs to the Right" option for scene tabs.

**Comment by Jamsers** [comment]
>This proposal is basically equivalent of "Close Tabs to the Right" option we have in scene tabs.

Right, one could argue by implementing this we'd be making the tab controls consistent between scene tabs and script tabs 😅

**Comment by PrecisionRender** [comment]
> > This proposal is basically equivalent of "Close Tabs to the Right" option we have in scene tabs.
> 
> Right, one could argue by implementing this we'd be making the tab controls consistent between scene tabs and script tabs 😅

Not sure why most don't like this idea. It seems like a great addition for consistency. I would find this useful myself.
